expression/column.go

```diff
@@ -124,6 +124,11 @@ func (col *CorrelatedColumn) IsCorrelated() bool {
 	return true
 }
 
+// ConstItem implements Expression interface.
+func (col *CorrelatedColumn) ConstItem() bool {
+	return false
```

Comment:
We just set it to true?

Reply:
I'm afraid it would cause some other problems.
For example, when building some scalar function like `from_unixtime(CorCol)`, we may use it to decide if we can evaluate the argument at that time.
If it is true, we may call `CorCol.Eval(NullRow)` and this function call is invalid.

Reply:
In my opinion, `ConstItem` means 1) this expression has no column reference, 2) this expression doesn't include some special scalar functions like `Rand` and `SysDate`.
`CorrelatedColumn` breaks the rule 1).